crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,7 +285,93 @@ impl Executor {
+            // drain the receiver channel
+            while let Ok(seq_action) = self.serialized_rollup_transactions.try_recv() {
+                let rollup_id = seq_action.rollup_id;
+                if let Err(e) = bundle_factory.try_push(seq_action) {
+                    warn!(
+                        rollup_id = %rollup_id,
+                        error = &e as &StdError,
+                        "failed to bundle sequence action: too large. sequence action is dropped."
+                    );
+                }
+            }
+
+            info!("draining remaining bundles from bundle factory");
+            // when shutting down, drain all the remaining bundles and submit to the sequencer
+            // to avoid any bundle loss.
+            let mut bundles_to_drain: Vec<SizedBundle> = vec![];
+            // drain all the bundles from the factory
+            loop {
+                let bundle = bundle_factory.pop_now();
+                if bundle.is_empty() {
+                    break;
+                }
+                bundles_to_drain.push(bundle);
+            }
+            let num_bundles_to_drain = bundles_to_drain.len();
+            if num_bundles_to_drain == 0 {
+                info!("no bundles to drain; executor shutdown tasks completed");
+                return Ok(());
+            }
+
+            info!(
+                "submitting {} bundles to sequencer to drain",
+                num_bundles_to_drain
+            );
+            // we need to submit the bundles sequentially since that is the guarantee
+            // the bundle factory gives to the end users.
+            for bundle in bundles_to_drain {
+                match self.submit_bundle(nonce, bundle).await {
+                    Ok(new_nonce) => {
+                        debug!("drained bundle successfully; new nonce: {}", new_nonce);
+                        nonce = new_nonce;
+                    }
+                    Err(e) => {
+                        error!(%e, "failed submitting bundle to sequencer during shutdown; \
+                                aborting shutdown");
+                        // if we can't drain a bundle after multiple retries, we can abort
+                        // the shutdown process
+                        return Err(e);
+                    }
+                }
+            }
+            info!("drained {} bundles successfully", num_bundles_to_drain);
+
+            Ok(())
+        };
+
+        match tokio::time::timeout(Duration::from_secs(13), shutdown_logic).await {
+            Ok(res) => {
+                if let Err(e) = res {
+                    error!(%e, "executor shutdown tasks aborted");
+                } else {
+                    info!("executor shutdown tasks completed successfully");
+                }
+            }
+            Err(err) => {
+                error!(%err, "executor shutdown tasks failed to complete in time");
```

Comment:
Yeah my initial thought is that i don't see the point of displaying the bundles? plus a bundle is a bunch of txs which wouldn't make sense displayed on the screen. I do not see the value of logging or tracking that, since we cannot as of now retry any submission offchain with that info.
I ll spend some more time thinking of it, but i don't see any strong reason as of now to display the missed bundles rather than just mentioning the number of bundles missed which could be good info, since that would inform us of the backpressure in executor.

Reply:
i was able to get it to report the bundles which weren't drained for now. if in prod, we see that it makes the logs unreadable or so, i can remove it!